internal/tiltfile/tiltfile_state.go

```diff
@@ -610,18 +679,56 @@ func (s *tiltfileState) assembleK8sV2() error {
 			r.resourceDeps = opts.resourceDeps
 			if opts.newName != "" && opts.newName != r.name {
 				if _, ok := s.k8sByName[opts.newName]; ok {
-					return fmt.Errorf("k8s_resource at %s specified to rename '%s' to '%s', but there is already a resource with that name", opts.tiltfilePosition.String(), r.name, opts.newName)
+					return fmt.Errorf("k8s_resource at %s specified to rename %q to %q, but there is already a resource with that name", opts.tiltfilePosition.String(), r.name, opts.newName)
 				}
 				delete(s.k8sByName, r.name)
 				r.name = opts.newName
 				s.k8sByName[r.name] = r
 			}
+
+			selectors := make([]k8sObjectSelector, len(opts.objects))
+			for i, o := range opts.objects {
+				s, err := selectorFromString(o)
+				if err != nil {
+					return errors.Wrapf(err, "Error making selector from string %q", o)
+				}
+				selectors[i] = s
+			}
+
+			for i, o := range opts.objects {
+				entities, ok := fragmentsToEntities[o]
+				if !ok || len(entities) == 0 {
+					return fmt.Errorf("No object identified by the fragment %q could be found. Unique fragments are: %s", o, strings.Join(uniqueFragmentNames, ", "))
+				}
+				if len(entities) > 1 {
+					matchingObjects := make([]string, len(entities))
+					for i, e := range entities {
+						matchingObjects[i] = fullNameFromK8sEntity(e)
+					}
+					return fmt.Errorf("%q is not a unique fragment. Objects that match %q are %s", o, o, strings.Join(matchingObjects, ", "))
+				}
+
+				entitiesToRemove := filterEntitiesBySelector(s.k8sUnresourced, selectors[i])
+				if len(entitiesToRemove) == 0 {
+					remainingUnresourcedFragments := make([]string, len(s.k8sUnresourced))
+					for i, entity := range s.k8sUnresourced {
+						remainingUnresourcedFragments[i] = fullNameFromK8sEntity(entity)
+					}
+					return fmt.Errorf("No object identified by the fragment %q could be found in remaining YAML. Valid remaining fragments are: %s", o, strings.Join(remainingUnresourcedFragments, ", "))
+				}
+				if len(entitiesToRemove) > 1 {
```

Comment:
I thiiiink this will never happen (🤞 ) b/c we checked above if a selector refers to more than one entity -- maybe panic? at least mention in a comment that we don't expect this to ever happen